samcli/yamlhelper.py

```diff
@@ -3,6 +3,7 @@
 """
 
 import json
+from collections import OrderedDict
```

Comment:
I am not sure if this will be a drop-in replacement. There could be code somewhere that assumes this is a pure dict `val isinstance dict` that might break because of this change. 

Also, in Py3.6 dicts are already in insertion order (similar to OrderedDict). While this is a problem for Py2.7 users, Py27 is dying very very soon and we should probably be migrating to Py36+. I am not sure if this worth the change now.

Do you have specific use cases that won't work without this change?

Reply:
I cannot think of any special usecases.
@paoptu023  What are your thoughts? 

Also, @sanathkr when are you planning to migrate to Py36+?

Reply:
I'm ok with not having this change if we will drop Py2.7 support soon. Otherwise, since we've already made the same change in SAR and aws cli, it might cause confusion due to the inconsistent behavior.